File: waysToScore/waysToScore.py
def waysToScorePoints(totalPoints, possiblePoints):
    if totalPoints < 0 or len(possiblePoints) == 0:
        return 0
    elif totalPoints == 0:
        return 1

    currentPoints = possiblePoints[0]

    return waysToScorePoints(totalPoints - currentPoints, possiblePoints) + \
            waysToScorePoints(totalPoints, possiblePoints[1:])

def waysToScore(totalPoints):
    return waysToScorePoints(totalPoints, [6, 3, 2])

testValues = [5, 12, 6, 11, 59, 24]

for testValue in testValues:
    print(f'waysToScore({testValue})\n{waysToScore(testValue)}\n', end='')

File: waysToScore/waysToScore.go
package main

import "fmt"

func waysToScorePoints(totalPoints int, possiblePoints []int) int {
    if totalPoints < 0 || len(possiblePoints) == 0 {
        return 0
    } else if totalPoints == 0 {
        return 1
    }

    currentPoints := possiblePoints[0]

    return waysToScorePoints(totalPoints - currentPoints, possiblePoints) +
            waysToScorePoints(totalPoints, possiblePoints[1:])
}

func waysToScore(totalPoints int) int {
    possiblePoints := []int{6, 3, 2}

    return waysToScorePoints(totalPoints, possiblePoints)
}

func main() {
    testValues := [6]int{5, 12, 6, 11, 59, 24}

    for _, testValue := range testValues {
        fmt.Printf("waysToScore(%d)\n%d\n", testValue, waysToScore(testValue))
    }
}
